### PR TITLE
Feature Flags component

### PR DESCRIPTION
**Pull request by jaketrent**
Resolves: #745 

Replaces: #756 

Reasoning for this solution in #756 comments.

### CODE REVIEW

File: packages/featureflags/react.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/// <reference types="react" />
```
Comment (jaketrent) [inline packages/featureflags/react.d.ts]: I want to add these, they make sense.  But I'm not a Typescript compiler.  How do we verify these are correct?

Comment (danethurber) [reply packages/featureflags/react.d.ts]: that's the question? we'd have to have some way to validation it. maybe something exists we can integrate?

**Comment by jaketrent** [comment]
Any concern about the reasoning that got us here?  Or the general approach?

**Comment by danethurber** [comment]
@jaketrent i love the simplicity and do think that other teams would find by using this approach

for the new colors...will we need to write a guide about how to provide the context value. we'll need to ensure that value matches up with what our button/link/etc components expect. does our team own that flag in launch darkly?

**Pull request merged**